src/languageservice/services/yamlSchemaService.ts

```diff
@@ -282,6 +293,28 @@ export class YAMLSchemaService extends JSONSchemaService {
             return resolveSchema();
         }
     }
+    
+    /**
+     * Retrieve schema if declared as modeline
+     * @param doc 
+     */
+    private getSchemaFromModeline(doc: any) : string{
+        if (doc instanceof SingleYAMLDocument) {
+            const modelineDeclaration = '# yaml-language-server:';
```

Comment:
What if I put 2 spaces between `#` and `yaml-language-server:`? Do we want it to be this strict? 

I wonder if tokenizing the line with something like  `string.split("/\b\s+(?!$)/");` would result on a more flexible implementation.

Reply:
I went to the easiest. it is also easier to read code. (for instance, I don't see the regexp that you provided to fit what is expected but i'm not sure if it is just an example or not)
That's possible to use regular expression. It allows to be more flexible for the users.
Not sure it is really needed.
I propose to report an issue to handle it in a separate PR. because there is the case of the double-space between `#` and `yaml-language-server` but there is also tabs. and if it is several spaces between the modeline attributes. and surely a bunch of other cases.

Reply:
`string.match(/\S+/g)` would be the way to get non-whitespace  and it is a loop from there. 

We should build this flexibility because at the current implementation if somebody tries to use the feature and misses a single space we fail silently. Or we can build a way to warn the users of such "syntax" errors when we detect the intention but that is essentially the same implementation.

Reply:
as always not so trivial with regexp. `\S` is matching also new line and page break Which we don't want here.

that said, given that we are providingline by line, I think we won't have the false-positives thanks to that.

Reply:
updated to cover more cases with extra spaces